src/link.js

```diff
@@ -46,22 +46,82 @@ export default class Link extends Plugin {
 		this.editor.editing.view.addObserver( ClickObserver );
 
 		/**
-		 * Balloon panel view to display the main UI.
+		 * The form view inside {@link #balloonPanelView}.
 		 *
-		 * @member {module:link/ui/balloonpanel~BalloonPanelView}
+		 * @member {module:link/ui/linkformview~LinkFormView}
 		 */
-		this.balloonPanelView = this._createBalloonPanel();
+		this.formView = this._createForm();
 
 		/**
-		 * The form view inside {@link #balloonPanelView}.
+		 * Contextual balloon plugin instance.
 		 *
-		 * @member {module:link/ui/linkformview~LinkFormView}
+		 * @private
+		 * @member {module:ui/contextualballoon~ContextualBalloon}
 		 */
-		this.formView = this._createForm();
+		this._balloon = this.editor.plugins.get( ContextualBalloon );
```

Comment:
This plugin manages contextual balloons and for me it doesn't have to leak out of this plugin.